flang/include/flang/Semantics/symbol.h

```diff
@@ -506,7 +506,10 @@ class Symbol {
       LocalityShared, // named in SHARED locality-spec
       InDataStmt, // initialized in a DATA statement, =>object, or /init/
       InNamelist, // in a Namelist group
-      CompilerCreated,
+      CompilerCreated, // A compiler created symbol
+      // For compiler created symbols that are constant but cannot legally have
+      // the PARAMETER attribute.
+      ReadOnly,
```

Comment:
I think we could fine tune this name just slightly to help clarify how it is intended to be used. It looks like `ReadOnly` is really only meant to be used in combination with compiler created artifacts. Since I didn't really follow what "cannot legally" was referring to, it would suggest more to me if this was called `CompilerCreatedReadOnly` (or `CompilerCreatedConstant`?).